src/app.ts

```diff
@@ -183,9 +187,47 @@ export async function newApp(
   registerAPIRoutes(app, APIBasePath, allowNotifyIPSourceRange);
   registerPagoPARoutes(app, PagoPABasePath, allowPagoPAIPSourceRange);
 
+  const timer = idpMetadataUpdater(app);
+  app.on("server:stop", () => {
+    clearInterval(timer);
+  });
   return app;
 }
 
+export function idpMetadataUpdater(
+  app: Express,
+  listener?: () => void
+): NodeJS.Timer {
+  return setInterval(async () => {
```

Comment:
I suggest to extract the async function to avoid having such a large lambda passed to `setInterval`

Reply:
fixed in 82e8752